src/gsy_e/models/strategy/energy_parameters/heatpump/heat_pump.py

```diff
@@ -408,29 +428,29 @@ def _update_last_time_slot_data(self, time_slot: DateTime):
         if last_time_slot not in self._source_temp_C.profile:
             return
         self.combined_state.increase_tank_temp_from_bought_energy(
-            last_time_slot, self._bought_energy_kWh, self._source_temp_C.profile[time_slot]
+            last_time_slot, self._bought_energy_kWh, self._source_temp_C.profile[last_time_slot]
         )
         self._bought_energy_kWh = 0.0
-        self._state.tanks.update_tanks_temperature(last_time_slot)
+        self._state.tanks.update_tanks_temperature(time_slot)
```

Comment:
I do not agree on this. Why are you updating the tank temperatures before the trading?